File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/generator/GeneratorNAtomicWith2SIn.java
package tasks.generator;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Int32;

@JavaTaskInfo(atomic = false, generator = true)
public class GeneratorNAtomicWith2SIn extends JavaTask {

	@InputPortInfo(name = "In1", dataType = Int32.class)
	public InputPort in1;
	@InputPortInfo(name = "In2", dataType = Int32.class)
	public InputPort in2;
	@OutputPortInfo(name = "Out", dataType = Int32.class)
	public OutputPort out;

	private int left  = -1;
	private int right = -2;
	
	@Override
	public void executeAsync(InputPort asynchIn) {
		if("In1".equals(asynchIn.getName())) left  = asynchIn.getData(Int32.class).getValue();
		if("In2".equals(asynchIn.getName())) right = asynchIn.getData(Int32.class).getValue();
	}

	@Override
	public void generate() {
		if(left == right) out.putData(new Int32(left));
		sleep(200);
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/generator/GeneratorNAtomicWith2SIn.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Int32 import Int32

@PythonTaskInfo(atomic=False, generator=True)
class GeneratorNAtomicWith2SIn(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Int32)
        self.in2 = InputPort(name="in2", data_type=Int32)
        self.out = OutputPort(name="out", data_type=Int32)

    def __init__(self):
        self.left = -1
        self.right = -2

    def execute_async(self, async_in):
        if async_in is self.in1:
            self.left  = async_in.get_data(Int32).values[0]
        elif async_in is self.in2:
            self.right = async_in.get_data(Int32).values[0]

    def generate(self):
        if self.left == self.right:
            self.out.put_data(Int32(self.left))
        self.sleep(200)
